### ISSUE TITLE
Panic with index out of range during compaction

### ISSUE THREAD

**Issue by sykesm** [cockroachdb/pebble#1669]
While ingesting several GB of data into Pebble, we encountered a panic during compaction. This was observed multiple times over the course of a few minutes.

```
panic: runtime error: index out of range [8936] with length 8936

goroutine 439372 [running]:
github.com/cockroachdb/pebble/internal/manifest.(*L0Sublevels).InitCompactingFileInfo(0xc0437d9080, {0xc01cdddd10, 0x1, 0xcb2db3b32d54bb80?})
        /home/sykesm/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20220322140431-c50a066abbd3/internal/manifest/l0_sublevels.go:367 +0x45e
github.com/cockroachdb/pebble.(*DB).removeInProgressCompaction(0xc000065800, 0xc000e21900)
        /home/sykesm/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20220322140431-c50a066abbd3/compaction.go:1264 +0x315
github.com/cockroachdb/pebble.(*DB).compact1(0xc000065800, 0xc000e21900, 0x0)
        /home/sykesm/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20220322140431-c50a066abbd3/compaction.go:1937 +0x5e7
github.com/cockroachdb/pebble.(*DB).compact.func1({0xfb0be0, 0xc02a2721e0})
        /home/sykesm/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20220322140431-c50a066abbd3/compaction.go:1873 +0xad
runtime/pprof.Do({0xfb0b70?, 0xc0000360f0?}, {{0xc00022c220?, 0x1000000010000?, 0xc0021a5760?}}, 0xc0021a5788)
        /usr/local/go/src/runtime/pprof/runtime.go:40 +0xa3
github.com/cockroachdb/pebble.(*DB).compact(0xc0434b05a0?, 0x0?, 0xc0021a5798?)
        /home/sykesm/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20220322140431-c50a066abbd3/compaction.go:1870 +0x6b
created by github.com/cockroachdb/pebble.(*DB).maybeScheduleCompactionPicker
        /home/sykesm/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20220322140431-c50a066abbd3/compaction.go:1666 +0x5ea
```

We were writing data at a rate of about 20 batches per second with each batch containing about 500 write operations for several hours before this occurred. I'm not sure what other information would be helpful.

The version of Pebble we are using is commit cockroachdb/pebble@c50a066abbd3.

**Comment by jbowens** [comment]
Hi @sykesm, thanks for the report. Could you provide us a copy with the manifest file(s) (eg, `MANIFEST-XXXXXX`) from the store directory? You can email them to me at `jackson` at cockroachlabs.com.

**Comment by jbowens** [comment]
Another question, do you use a custom comparator and if so, can you provide its implementation?

**Comment by bananabrick** [comment]
@sykesm Manifest might still be useful. We also keep around older manifests, so you'll probably see the older manifest in the directory.

**Comment by nicktrav** [comment]
Fixed in #1784.
